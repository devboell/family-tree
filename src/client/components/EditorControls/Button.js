import styled from 'styled-components'


const Button = styled.button`
  padding: 3px 10px;
  margin-right: 20px;
  font-size: 12px;
  cursor: pointer;
`

export default Button
