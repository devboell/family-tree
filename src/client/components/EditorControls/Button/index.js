import styled from 'styled-components'


const Button = styled.button`
  padding: 3px 10px;
  margin-right: 20px;
  font-size: 12px;
`

export default Button
