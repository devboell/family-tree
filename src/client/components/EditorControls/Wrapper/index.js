import styled from 'styled-components'


const Wrapper = styled.div`
  display: flex;
  border: white solid 2px;
  border-radius: 5px;
  padding: 5px 10px;
`

export default Wrapper
