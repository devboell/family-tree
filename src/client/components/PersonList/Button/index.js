import styled from 'styled-components'


const Wrapper = styled.button`
  width: 100%;
  background-color: rgb(76, 175, 80);
  border: none;
  color: white;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 12px;

  &:hover {
    background-color: gray;
  }
`

export default Wrapper
