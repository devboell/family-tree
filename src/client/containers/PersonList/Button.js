import styled, { css } from 'styled-components'

const hover = css`
  background-color: gray;
`

const hoverAndSelected = css`
  background-color: rgb(18, 106, 22);
  cursor: default;
`

const Button = styled.button`
  width: 100%;
  background-color: ${props => (
    props.isSelected
    ? 'rgb(18, 106, 22)'
    : 'rgb(76, 175, 80)'
  )};
  border: none;
  color: white;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 12px;
  outline: 0;
  cursor: pointer;

  &:hover {
    ${props => (
      props.isSelected
      ? hoverAndSelected
      : hover
    )};
  }
`
Button.displayName = 'Button'

export default Button
