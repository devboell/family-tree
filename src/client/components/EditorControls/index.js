import React from 'react'
import { PropTypes } from 'prop-types'

import { editModes } from '../../containers/EditorPage/constants'

import Wrapper from './Wrapper'
import Button from './Button'

const EditorControls = ({
  editMode,
  selectedPerson,
  onCreatePerson,
  onRemovePerson,
  removePersonMutation,
}) =>
  <Wrapper>
    <Button onClick={onCreatePerson}>New</Button>
    {editMode === editModes.UPDATE
      ?
        <Button
          onClick={
            () => {
              removePersonMutation({ id: selectedPerson.id })
              .then(onRemovePerson())
            }
          }
        >
          Remove
        </Button>
      : null
    }
  </Wrapper>

EditorControls.propTypes = {
  editMode: PropTypes.string.isRequired,
  selectedPerson: PropTypes.shape({}).isRequired,
  onCreatePerson: PropTypes.func.isRequired,
  onRemovePerson: PropTypes.func.isRequired,
  removePersonMutation: PropTypes.func.isRequired,
}

export default EditorControls
