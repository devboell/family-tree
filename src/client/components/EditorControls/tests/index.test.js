import React from 'react'
import EditorControls from '../index'
import Button from '../Button'

const props = {
  editMode: 'create',
  selectedPerson: { id: 1 },
  onCreatePerson: jest.fn(),
  onRemovePerson: jest.fn(),
  removePersonMutation: jest.fn(() => new Promise(
    reslove => reslove(),
  )),
}

describe('EditorControls component', () => {
  it('renders in create mode', () => {
    const wrapper = shallow(<EditorControls {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders in update mode', () => {
    const updateProps = { ...props, editMode: 'update' }
    const wrapper = shallow(<EditorControls {...updateProps} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('calls removePersonMutation', () => {
    const updateProps = { ...props, editMode: 'update' }
    const wrapper = shallow(<EditorControls {...updateProps} />)
    wrapper.find(Button).at(1).simulate('click')

    expect(props.removePersonMutation).toHaveBeenCalledWith({ id: 1 })
    expect(props.onRemovePerson).toHaveBeenCalled()
  })
})
