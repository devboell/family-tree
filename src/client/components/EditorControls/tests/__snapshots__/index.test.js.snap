// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorControls component renders in create mode 1`] = `
<styled.div>
  <styled.button
    onClick={[Function]}
  >
    New
  </styled.button>
</styled.div>
`;

exports[`EditorControls component renders in update mode 1`] = `
<styled.div>
  <styled.button
    onClick={[Function]}
  >
    New
  </styled.button>
  <styled.button
    onClick={[Function]}
  >
    Remove
  </styled.button>
</styled.div>
`;
